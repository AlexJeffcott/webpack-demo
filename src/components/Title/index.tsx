import React, { FC } from 'react'
// @ts-ignore
import styles from './title.module.scss'

type Props = { title: string }
const Title: FC<Props> = ({ title }) => {
  return (
    <h1 data-testid="Title Component" className={styles.title}>
      {title}
    </h1>
  )
}

export default Title
