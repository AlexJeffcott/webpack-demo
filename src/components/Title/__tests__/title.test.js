import React from 'react'
import { cleanup, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Title from '../'

describe(`Title Component`, () => {
  afterEach(cleanup)

  const testData = [
    [
      'has correct simple title text',
      {
        title: `Hello React!`,
      },
      `Hello React!`,
    ],
    [
      'has correct tricky title text',
      {
        title: `abcdefghijklmnopqrstuvwxyz1234567890!~#@?,.£$%^&*()-=+|/'"äüößú`,
      },
      `abcdefghijklmnopqrstuvwxyz1234567890!~#@?,.£$%^&*()-=+|/'"äüößú`,
    ],
  ]

  test.each(testData)(`%p`, (testTitle, testParams, testExpected) => {
    const { asFragment, getByTestId } = render(<Title {...testParams} />)
    expect(getByTestId('Title Component')).toHaveTextContent(testExpected)
    expect(asFragment()).toMatchSnapshot()
  })
})
