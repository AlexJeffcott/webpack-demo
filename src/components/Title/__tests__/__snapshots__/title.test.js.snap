// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component "has correct simple title text" 1`] = `
<DocumentFragment>
  <h1
    data-testid="Title Component"
  >
    Hello React!
  </h1>
</DocumentFragment>
`;

exports[`Title Component "has correct tricky title text" 1`] = `
<DocumentFragment>
  <h1
    data-testid="Title Component"
  >
    abcdefghijklmnopqrstuvwxyz1234567890!~#@?,.£$%^&*()-=+|/'"äüößú
  </h1>
</DocumentFragment>
`;
